typescript/cli/src/config/chain.ts

```diff
@@ -38,19 +39,27 @@ export async function createChainConfig({
   context: CommandContext;
 }) {
   logBlue('Creating a new chain config');
-  const name = await input({
-    message: 'Enter chain name (one word, lower case)',
-  });
-  const chainId = await input({ message: 'Enter chain id (number)' });
-  const domainId = chainId;
+
   const rpcUrl = await input({ message: 'Enter http or https rpc url' });
+  const network = await new ethers.providers.JsonRpcProvider(
+    rpcUrl,
+  ).getNetwork();
+
+  let { chainId, name } = network;
+  if (name === 'unknown') {
+    name = await input({
+      message: 'Enter chain name (one word, lower case)',
```

Comment:
possible nit: does input auto-add a colon at the end here?

Reply:
idk but would prefer not to change this everywhere

Reply:
Should we really rely on the name provided by the network given that we've invented our own naming type:
https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/main/typescript/sdk/src/metadata/customZodTypes.ts#L20